web/static/js/components/activity/ActivityDescription.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { Component, PropTypes } from 'react'
+import { translate } from 'react-i18next'
+
+class ActivityDescription extends Component {
+  text(activity) {
+    const { t } = this.props
+    const metadata = activity.metadata
+    let text, reportType
+    switch (activity.entityType) {
+      case 'survey':
+        switch (activity.action) {
+          case 'download':
+            reportType = activity.metadata['reportType'].split('_').join(' ')
+            text = t('Downloaded <i>{{surveyName}}</i> {{reportType}}', {surveyName: metadata['surveyName'], reportType: reportType})
```

Comment:
This method could be much shorter if you just `return` each string and build the `{__html: text}` object in the `render` method